api/v1alpha2/linodefirewall_types.go

```diff
@@ -0,0 +1,188 @@
+/*
+Copyright 2023 Akamai Technologies, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha2
+
+import (
+	"github.com/linode/linodego"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+)
+
+const (
+	// FirewallFinalizer allows ReconcileLinodeFirewall to clean up Linode resources associated
+	// with LinodeFirewall before removing it from the apiserver.
+	FirewallFinalizer = "linodefirewall.infrastructure.cluster.x-k8s.io"
+)
+
+// LinodeFirewallSpec defines the desired state of LinodeFirewall
+type LinodeFirewallSpec struct {
+	// +optional
+	FirewallID *int `json:"firewallID,omitempty"`
+
+	// +optional
+	// +kubebuilder:default=false
+	Enabled bool `json:"enabled,omitempty"`
+
+	// +kubebuilder:validation:MinLength=3
+	// +kubebuilder:validation:MaxLength=63
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="Value is immutable"
+	// +optional
+	Label string `json:"label,omitempty"`
```

Comment:
question: what do we use this label for? Custom naming for the firewall?

Reply:
I think that's leftover from March and can probably be removed. IIRC we make the Linode label based on the k8s resource name